test/integration/targets/ec2_instance/tasks/iam_instance_role.yml

```diff
@@ -77,26 +66,34 @@
     - name: Update instance with new instance_role
       ec2_instance:
         name: "{{ resource_prefix }}-test-instance-role"
-        image_id: "{{ ec2_ami_image[aws_region] }}"
+        image_id: "{{ ec2_ami_image }}"
         security_groups: "{{ sg.group_id }}"
-        instance_type: t2.micro
+        instance_type: "{{ ec2_instance_type }}"
         instance_role: "{{ iam_role_2.arn.replace(':role/', ':instance-profile/') }}"
-        <<: *aws_connection_info
       register: instance_with_updated_role
-      until: instance_with_updated_role is not failed
-      retries: 10
+
+    # XXX We shouldn't need this
```

Comment:
The associated instance role not updating immediately isn't an error I've seen crop up, but I assume you have?

Reply:
It popped up once in the 10 or so runs I tried